src/compiler/scala/tools/nsc/typechecker/Typers.scala

```diff
@@ -4614,7 +4614,7 @@ trait Typers extends Adaptations with Tags with TypersTracking with PatternTyper
         case tp@(MethodType(_, _) | PolyType(_, MethodType(_, _))) => // (1)
           val formals = tp.params
           if (isFunctionType(pt) || samMatchesFunctionBasedOnArity(samOf(pt), formals)) methodValue
-          else adapt(methodValue, mode, checkArity(methodValue)(functionTypeWildcard(formals.length)), original)
+          else adapt(methodValue, mode, checkArity(methodValue)(functionTypeWildcard(formals.length), formals.length), original)
```

Comment:
Not sure if `formals.length` is correct here?

Reply:
I think it is, but with the caveat about the "given" message.